### ISSUE TITLE
sql: schema change leads to "not hit TTL deadline" error

### ISSUE THREAD

**Issue by andreimatei** [cockroachdb/cockroach#23251]
Start a server, run:
```
create table foo(k int primary key); drop table foo; create table foo(k int primary key); drop table foo;
```

Around 40s later, the log periodically starts showing:
```
W180228 22:04:04.082715 152 sql/schema_changer.go:1146  [n1] Error executing schema change: not hit TTL deadline: -24h59m31.392735s remaining
```

@vivekmenezes, do you know what this is about?

**Comment by vivekmenezes** [comment]
That's just saying that the data is waiting for the GC interval to expire
before getting cleared. But let me fix the error message and let me block
this message from being logged

**Comment by jordanlewis** [comment]
+1, hitting this a lot and it's confusing.

**Comment by tschottdorf** [comment]
Also if that message is spammy, consider throwing a `log.Every` in there too.

**Comment by jordanlewis** [comment]
The fix for this isn't in release-2.0, is that problematic?
